File: currency_exchange/currency_exchange.py
def currency_exchange():
    count = 0
    for coin500 in range(0, 3):
        for coin100 in range(0, 11):
            for coin50 in range(0, 16):
                for coin10 in range(0, 16):
                    if coin500 + coin100 + coin50 + coin10 <= 15:
                        if coin500 * 500 + coin100 * 100 + coin50 * 50 + coin10 * 10 == 1000:
                            count += 1
    print count

currency_exchange()

File: currency_exchange/currency_exchange.js
var CurrencyExchange = {
    count: 0,
    start: function() {
        for (var coin500 = 0; coin500 <= 2; coin500++) {
            for (var coin100 = 0; coin100 <= 10; coin100++) {
                for (var coin50 = 0; coin50 <= 15; coin50++) {
                    for (var coin10 = 0; coin10 <= 15; coin10++) {
                        if (coin500 + coin100 + coin50 + coin10 <= 15) {
                            if (coin500 * 500 + coin100 * 100 + coin50 * 50 + coin10 * 10 == 1000) {
                                this.count++;
                            }
                        }
                    }
                }
            }
        }
    }
};

CurrencyExchange.start();
console.log(CurrencyExchange.count);